runtime/src/main/java/org/corfudb/runtime/object/DiskBackedSMRSnapshot.java

```diff
@@ -134,4 +152,12 @@ public <K, V> RocksDbEntryIterator<K, V> newIterator(ISerializer serializer, Tra
             return iterator;
         });
     }
+
+    private void incrementGauge(String name, ConcurrentHashMap<UUID, AtomicLong> map, UUID objectId) {
+        map.computeIfAbsent(objectId, id -> {
+            final AtomicLong counter = new AtomicLong();
+            MicroMeterUtils.gauge(name, counter, AtomicLong::get, OBJECT_ID_TAG, objectId.toString());
```

Comment:
I'm not sure how useful these metrics will be. You can get an estimation of unreleased snapshot count for each table, but that info is not enough to be a leakage indicator - you can have a spike of unreleased snapshots due to more updates in that time window.

Reply:
I think it's still an estimation because metrics are not taken at exactly T+N min. Perhaps a more reliable way is to maintain a snapshot creation time and report the 'age' of snapshots at metrics reading.

Reply:
The issue with the create/release counters is, you can't really tell for sure if there is a leak - imagine that due to a few millisecond time spew in metrics reading, there are 10 readings where num of creation > num of release, and  another 10 readings where num of create < num of release. On the contrary, a metrics that reports N snapshots being alive for more than 2x expiration duration can clearly indicate that there is a leak.

Reply:
No I'm following your comment `If the number of snapshots released at T + N minutes is less than snapshots created at time T`. In my example, T is T00:01:00, N is 5 mins, and the second reading at T+N has a few seconds skew which is normal. `If the number of snapshots released at T + N minutes` is 55 and `snapshots created at time T` is 50. [T+N, T+N+skew] has released the additional 5 snapshots.

Reply:
Yes Vjeko. That's a good example. I'm comparing Created@T and Released@(T+N) like you suggested previously. When you have such reports, you can't tell much from comparing Created@T1=50 and Released@T2=55. 
```
T1: Metrics Report (Created = 50, Released = 0)
T2: Metrics Report (Created = 55, Released = **55**)
```
BTW I understand that `Number of snapshots created always has to be equal or greater than snapshots released at any point in time.` is mostly true under the assumption that the create and release gauges are read at the same time atomically. A counter example can be that in one metrics report, after create gauge is read, a new snapshot is created and released before the release gauge is read, though it's mostly just for the theory.